src/Component/src/Doctrine/Common/State/RemoveProcessor.php

```diff
@@ -34,8 +36,12 @@ public function process(mixed $data, Operation $operation, Context $context): mi
             return null;
         }
 
-        $manager->remove($data);
-        $manager->flush();
+        try {
+            $manager->remove($data);
+            $manager->flush();
+        } catch (ForeignKeyConstraintViolationException) {
+            throw new DeleteHandlingException();
```

Comment:
I'm wondering if this could be better to introduce a new exception that would have a better name and less proprerties.

Reply:
👍 Extracting it from Core would be great, as it's pretty low level and the class really shouldn't originate that high up.